lavalink/Client.py

```diff
@@ -123,18 +114,34 @@ def unregister_hook(self, func):
         """ Updates a player's state when a payload with opcode ``playerUpdate`` is received. """
         guild_id = int(data['guildId'])
 
-        if guild_id in self.players:
-            player = self.players.get(guild_id)
-            player.position = data['state'].get('position', 0)
-            player.position_timestamp = data['state']['time']
+        for node in self.nodes:
+            if guild_id in node.players:
+                player = node.players.get(guild_id)
+                player.position = data['state'].get('position', 0)
+                player.position_timestamp = data['state']['time']
 
     async def get_tracks(self, query):
         """ Returns a Dictionary containing search results for a given query. """
         log.debug('Requesting tracks for query {}'.format(query))
-
-        async with self.http.get(self.rest_uri + quote(query), headers={'Authorization': self.password}) as res:
+        await self.nodes.ready.wait()

```

Comment:
Perhaps this should timeout after a few seconds or so in case there are no nodes or they fail to connect (in the event the user's network goes down) so this isn't stuck waiting forever

Reply:
How should we handle timeouts? Or should we just throw the default timeout exception since this version of lavalink.py would be probably meant for advanced users

Reply:
Also how long should the timeout be? 10? 30 seconds?

Reply:
Throwing would be good but we could also handle it ourselves and return an empty list, although that may not be verbose enough and may lead to users asking why they're getting no results. I'm leaning towards the former (throw timeout)

Reply:
I'd say keep the timeout at about 10 seconds for now, *should* be enough in the event that a user is simply rebooting the server

Reply:
does it? hmmm i think only the websockets wait

Reply:
[Yep, just websockets](https://github.com/Devoxin/Lavalink.py/blob/master/lavalink/WebSocket.py#L38) although I don't think anything else needs to wait for bot to ready up as WebSocket is the only thing (that I can think of off the top of my head, at least) that requires bot properties that are only accessible once logged in

Reply:
Yup, but if the startup takes more then 10 seconds it WILL throw a timeout. One solution would be to wait for on_ready, which would be smart tbh

Reply:
NoNodesAvailable would be thrown only if we wouldn't use that ready.wait() there

Reply:
Probably a better alternative to potentially awaiting forever though

Reply:
yup, I'd wait for 20 sec then throw NoNodesAvailable